### PR TITLE
Add convenience APIs to get control flow graph

### LINKED ISSUE #27627: Add convenience APIs for control flow graph
1. `static SemanticModel.GetControlFlowGraph(IOperation rootOperation)` – expects rootOperation.Parent == null.
2. `static SemanticModel.GetEnclosingFlowGraph(IOperation operation)` – walks up to the root and invokes the above API
AND/OR
`IOperationExtensions.GetEnclosingFlowGraph(this IOperation operation)`
3. `OperationBlockAnalysisContext.GetControlFlowGraphs()`, `OperationBlockStartAnalysisContext.GetControlFlowGraphs()`, `OperationAnalysisContext.GetEnclosingFlowGraph()`

We all agreed upon 1. Need to finalize the remaining APIs to add.

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #27627

Pending:
1. Design discussion on the APIs
2. Add unit tests for approved APIs

### CODE REVIEW

**Comment by mavasani** [comment]
@AlekseyTs @gafter @tmat @DustinCampbell @jinujoseph @333fred @heejaechang This PR is now ready for review. It implements items 1 and 2 in the list below, remaining two will be in a separate PR:

1. Replace `static SemanticModel.GetControlFlowGraph(IOperation)` with `ControlFlowGraph.Create(IOperation)` factory methods.
2. Add `GetControlFlowGraph` APIs to all operation analysis contexts
3. Add `static ControlFlowGraph.Create(SyntaxNode, SemanticModel)` API
4. Expose `IOperation.SemanticModel` API

File: src/Compilers/Core/Portable/Compilation/SemanticModel.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@
 using System.Diagnostics;
 using System.Threading;
 using Microsoft.CodeAnalysis.FlowAnalysis;
+using Microsoft.CodeAnalysis.Operations;
```
Comment (333fred) [inline src/Compilers/Core/Portable/Compilation/SemanticModel.cs]: Unrelated?

**Comment by mavasani** [comment]
Ping @dotnet/roslyn-compiler for one more review.

**Comment by mavasani** [comment]
@333fred I will remove the unnecessary using in the next PR.

@AlekseyTs, that looks like a good idea, I will rebase the PR to master. Tagging @jinujoseph for ask mode approval

**Comment by jinujoseph** [comment]
Approved to merge for 15.8.Preview4
